### PR TITLE
update weights from disk

### PR DESCRIPTION
**Pull request by zhaochenyang20**
## Motivation

https://github.com/sgl-project/sglang/pull/2264

Split this huge PR into three smaller PRs.

## Modifications

This pr rename the `udpate_weights` API to `update_weights_from_disk`.

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/models/llama.py [hunk 1]
```diff
@@ -305,6 +305,14 @@ def __init__(
         self.lm_head = ParallelLMHead(config.vocab_size, config.hidden_size)
         self.logits_processor = LogitsProcessor(config)
         self.pooler = Pooler(pooling_type=PoolingType.LAST, normalize=True)
+        self.stacked_params_mapping = [
```
Comment (merrymercy) [inline python/sglang/srt/models/llama.py]: revert this?

File: python/sglang/srt/server.py [hunk 2]
```diff
@@ -269,6 +270,11 @@ async def stream_results() -> AsyncIterator[bytes]:
             )
 
 
+# fastapi implicitly converts json in the request to obj (dataclass)
+app.post("/generate")(generate_request)
```
Comment (merrymercy) [inline python/sglang/srt/server.py]: revert

File: python/sglang/srt/server.py [hunk 3]
```diff
@@ -938,3 +944,54 @@ def stop_profile(self):
 
     async def get_server_info(self):
         return await _get_server_info()
+
+    def generate(
```
Comment (merrymercy) [inline python/sglang/srt/server.py]: revert unrelated changes

File: python/sglang/srt/utils.py [hunk 4]
```diff
@@ -38,13 +39,24 @@
...
 import torch
+import torch.distributed
 import torch.distributed as dist
 import triton
 import zmq
 from fastapi.responses import ORJSONResponse
 from packaging import version as pkg_version
 from starlette.routing import Mount
 from torch import nn
+from torch.distributed.distributed_c10d import (
```
Comment (merrymercy) [inline python/sglang/srt/utils.py]: revert unrelated changes

**Review by merrymercy — CHANGES_REQUESTED** [review]
also update `/docs`

File: python/sglang/srt/managers/scheduler.py [hunk 5]
```diff
@@ -1363,9 +1363,9 @@ def abort_request(self, recv_req: AbortReq):
                     req.to_abort = True
                     break
 
-    def update_weights(self, recv_req: UpdateWeightReqInput):
+    def update_weights_from_disk(self, recv_req: UpdateWeightFromDiskReqInput):
         """In-place update of the weights."""
```
Comment (merrymercy) [inline python/sglang/srt/managers/scheduler.py]:
```suggestion
        """In-place update of the weights from disk."""
```

File: python/sglang/srt/server.py [hunk 6]
```diff
@@ -269,6 +269,9 @@ async def stream_results() -> AsyncIterator[bytes]:
             )
 
 
+app.put("/generate")(generate_request)
```
Comment (merrymercy) [inline python/sglang/srt/server.py]: remove this

**Review by merrymercy — CHANGES_REQUESTED** [review]
also update the `update_weights` in `native_api.ipynb`

**Review by merrymercy — APPROVED** [review]
